<launch>

  <include file="$(find ros_objslampp_ycb_video)/launch/singleview_3d_pose_estimation.launch">
  </include>

  <group ns="camera">
    <node name="octomap_server"
          pkg="ros_objslampp_ycb_video" type="octomap_server"
          clear_params="true"
          output="screen">
      <remap from="cloud_in" to="depth_registered/points" />
      <remap from="label_ins_in" to="mask_rcnn_instance_segmentation/output/label_ins" />
      <rosparam subst_value="true">
        resolution: 0.01
      </rosparam>
    </node>
    <node name="grids_to_mesh_markers"
          pkg="ros_objslampp_ycb_video" type="voxel_grids_to_mesh_markers.py"
          clear_params="true">
      <remap from="~input" to="octomap_server/output/grids" />
    </node>
    <node name="grids_to_markers"
          pkg="ros_objslampp_ycb_video" type="voxel_grids_to_markers.py"
          clear_params="true">
      <remap from="~input" to="octomap_server/output/grids" />
    </node>
    <node name="grids_noentry_to_markers"
          pkg="ros_objslampp_ycb_video" type="voxel_grids_to_markers.py"
          clear_params="true">
      <remap from="~input" to="octomap_server/output/grids_noentry" />
    </node>
  </group>

  <include file="$(find ros_objslampp_ycb_video)/launch/orb_slam2_d435_rgbd.launch">
  </include>

</launch>
